pkg/cmd/roachtest/cdc.go

```diff
@@ -120,26 +127,30 @@ func registerCDC(r *registry) {
 				case <-time.After(time.Second):
 				}
 
-				// Until we have changefeed monitoring, query the job progress
-				// proto directly.
-				var progressBytes []byte
+				var status string
+				var hwRaw gosql.NullString
 				if err := db.QueryRow(
-					`SELECT progress FROM system.jobs WHERE id = $1`, jobID,
-				).Scan(&progressBytes); err != nil {
+					`SELECT status, high_water_timestamp FROM crdb_internal.jobs WHERE job_id = $1`,
+					jobID,
+				).Scan(&status, &hwRaw); err != nil {
 					return err
 				}
-				var progress jobspb.Progress
-				if err := protoutil.Unmarshal(progressBytes, &progress); err != nil {
-					return err
+				if status != `running` {
+					return errors.Errorf(`unexpected status: %s`, status)
 				}
-				if hw := progress.GetHighWater(); hw != nil && hw.WallTime > 0 {
+				if hwRaw.Valid {
+					hwWallTime, err := strconv.ParseInt(
```

Comment:
There is a method "tree.DecimalToHLC" that parses this already. It might be preferable to parse the string into an apd.Decimal and then use DecimalToHLC to extract the timestamp.